parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -725,4 +735,29 @@ private static final int compareTwoByteArrays(byte[] array1, int offset1, int le
     else if (length1 < length2) { return 1;}
     else { return -1; }
   }
+
+  // Uses UTF8-aware byte comparisons, converting signed bytes to int so that ASCII range characters (bytes 0-127)
+  // will be "less than" other bytes, so comparison is lexicographic.
+  // See {@link BinaryData#compareBytes https://github.com/apache/avro/blob/master/lang/java/avro/src/main/java/org/apache/avro/io/BinaryData.java#L184}
+  // methods in Avro for another example of this.
+  private static final int compareTwoByteBuffersUTF8(ByteBuffer array1, int offset1, int length1,
+                                                     ByteBuffer array2, int offset2, int length2) {
+    if (array1 == null && array2 == null) return 0;
+    if (array1 == array2 && offset1 == offset2 && length1 == length2) return 0;
+    int min_length = (length1 < length2) ? length1 : length2;
+    for (int i = 0; i < min_length; i++) {
+      int value1 = array1.get(i + offset1) & 0xFF;
+      int value2 = array2.get(i + offset2) & 0xFF;
+      if (value1 < value2) {
```

Comment:
If I simplify these statements should I go ahead and make that change to all the compare* methods in this project?